### PR TITLE
[WIP] Multi-GPU training

### LINKED ISSUE #16: Multi-GPU synchronous training
Distributed training is currently asynchronous. It should also support a local multi-GPU synchronous mode.

### PR DESCRIPTION
**Pull request by guillaumekln**
This PR adds multi GPU training using a modified version of the [replicate_model_fn](https://www.tensorflow.org/versions/r1.5/api_docs/python/tf/contrib/estimator/replicate_model_fn) function.

Points to address:

* [x] Cleanup summaries scopes in TensorBoard
* [x] Re-add gradients summaries in TensorBoard (?)

Closes #16.

### CODE REVIEW

**Comment by guillaumekln** [comment]
This implementation works but does not give the expected results. Further analysis indicates that we should prefer in-graph replication only.
